cmd/snapshotgc/snapshotgc.go

```diff
@@ -0,0 +1,61 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	applicationapiv1alpha1 "github.com/redhat-appstudio/application-api/api/v1alpha1"
+	releasev1alpha1 "github.com/redhat-appstudio/release-service/api/v1alpha1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var (
+	scheme = runtime.NewScheme()
+)
+
+func init() {
+	applicationapiv1alpha1.AddToScheme(scheme)
+	clientgoscheme.AddToScheme(scheme)
+	releasev1alpha1.AddToScheme(scheme)
+}
+
+type snapshotData struct {
+	snapshot applicationapiv1alpha1.Snapshot
+	// environmentBinding applicationapiv1alpha1.SnapshotEnvironmentBinding
+	release releasev1alpha1.Release
+}
+
+func getSnapshotsForReleases(
+	cl client.Client,
+	snapToData map[string]snapshotData,
+) map[string]snapshotData {
+	releases := &releasev1alpha1.ReleaseList{}
+	err := cl.List(context.Background(), releases)
+	if err != nil {
+		fmt.Printf("Failed to list releases: %v\n", err)
+		os.Exit(1)
+	}
+
+	for _, release := range releases.Items {
+		snap := &applicationapiv1alpha1.Snapshot{}
+		err := cl.Get(context.Background(), types.NamespacedName{
+			Name:      release.Spec.Snapshot,
+			Namespace: release.Namespace,
+		}, snap)
+		if err != nil {
+			fmt.Printf("Failed Getting snapshot: %v\n", err)
+		}
+
+		data, ok := snapToData[snap.Namespace+"|"+snap.Name]
```

Comment:
This is to handle the same snapshot released in multiple releases?

Reply:
That snapshot data struct can hold references to both bindings and release. later on, we're going to have another (all too similar) function that will collect bindings. We're checking here if the snapshot was added already for bindings (or something else).